pkg/operator/controller/gatewayclass/istio.go

```diff
@@ -0,0 +1,209 @@
+		// which otherwise would create the default gatewayclass.
+		// However, Istio's deployment controller also uses this
+		// setting, namely by reconciling gateways that specify this
+		// gatewayclass (irrespective of the controller name that the
+		// gatewayclass specifies), so we don't want to leave this at
+		// the default value, which is "istio".  If we didn't set this,
+		// then our Istiod instance might try to reconcile gateways
+		// belonging to an unrelated Istiod instance.
+		"PILOT_GATEWAY_API_DEFAULT_GATEWAYCLASS_NAME": OpenShiftDefaultGatewayClassName,
+		// Watch Gateway API and Kubernetes resources in all namespaces,
+		// but ignore Istio resources that don't match our label
+		// selector.  (We do not specify the label selector, so this
+		// causes Istio to ignore all Istio resources.)
+		"PILOT_ENABLE_GATEWAY_CONTROLLER_MODE": "true",
+		// Only reconcile resources that are associated with
+		// gatewayclasses that have our controller name.
+		"PILOT_GATEWAY_API_CONTROLLER_NAME": OpenShiftGatewayClassControllerName,
+		// Don't create an "istio-remote" gatewayclass for
+		// "multi-network gateways".  This is an Istio feature that I
+		// haven't really found any explanation for.
+		"PILOT_MULTI_NETWORK_DISCOVER_GATEWAY_API": "false",
+	}
+	istio := sailv1.Istio{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:       name.Namespace,
+			Name:            name.Name,
+			OwnerReferences: []metav1.OwnerReference{ownerRef},
+		},
+		Spec: sailv1.IstioSpec{
+			Namespace: controller.DefaultOperandNamespace,
+			UpdateStrategy: &sailv1.IstioUpdateStrategy{
+				Type: sailv1.UpdateStrategyTypeInPlace,
+			},
+			Values: &sailv1.Values{
+				Global: &sailv1.GlobalConfig{
+					IstioNamespace: ptr.To(controller.DefaultOperandNamespace),
+					// Logging: &sailv1.GlobalLoggingConfig{
+					// 	Level: ptr.To("all:debug"),
+					// },
+				},
+				Pilot: &sailv1.PilotConfig{
+					Cni: &sailv1.CNIUsageConfig{
+						Enabled: ptr.To(false),
+					},
+					Enabled:            ptr.To(true),
+					Env:                pilotContainerEnv,
+					ExtraContainerArgs: []string{},
+				},
+				SidecarInjectorWebhook: &sailv1.SidecarInjectorConfig{
+					EnableNamespacesByDefault: ptr.To(false),
+				},
+				MeshConfig: &sailv1.MeshConfig{
+					AccessLogFile: ptr.To("/dev/stdout"),
+					DefaultConfig: &sailv1.MeshConfigProxyConfig{
+						ProxyHeaders: &sailv1.ProxyConfigProxyHeaders{
+							Server: &sailv1.ProxyConfigProxyHeadersServer{
+								Disabled: ptr.To(true),
+							},
+							EnvoyDebugHeaders: &sailv1.ProxyConfigProxyHeadersEnvoyDebugHeaders{
+								Disabled: ptr.To(true),
+							},
+							MetadataExchangeHeaders: &sailv1.ProxyConfigProxyHeadersMetadataExchangeHeaders{
+								Mode: sailv1.ProxyConfigProxyHeadersMetadataExchangeModeInMesh,
+							},
+						},
+					},
+					IngressControllerMode: sailv1.MeshConfigIngressControllerModeOff,
+				},
+			},
+			Version: "v1.24.3",
```

Comment:
Would it be worthwhile for us to conduct a brief review of the choice of this particular version?  Do we have the option of using a different version with servicemeshoperator3?

Reply:
I just picked the version that I knew OSSM 3 supported.  OSSM 3.0.0 [supports Istio 1.24.3](https://github.com/openshift-service-mesh/sail-operator/blob/3.0.0/api/v1/istio_types.go#L39-L40), which [has Gateway API v1.2.1](https://github.com/istio/istio/blob/1.24.3/go.mod#L108).  (Istio 1.24.2 [has some prerelease version](https://github.com/istio/istio/blob/1.24.2/go.mod#L108).)